HACKING.md

```diff
@@ -0,0 +1,409 @@
+# Developing Salt
+
+
+So you want to contribute to the Salt project? Excellent! You can help in a
+number of ways:
+
+- Using Salt and opening bug reports.
+- Joining a [working group](https://github.com/saltstack/community).
+- Answering questions on [irc][salt on freenode], the
+  [community Slack](https://saltstackcommunity.herokuapp.com/), or the
+  [salt-users mailing list](https://groups.google.com/forum/#!forum/salt-users).
+- Fixing issues.
+
+If you'd like to update docs or fix an issue, you're going to need the Salt
+repo. The best way to contribute is using [Git](https://git-scm.com/).
+
+## Environment Setup
+
+To hack on Salt or the docs you're going to need to setup your development
+environment. If you already have a workflow that you're comfortable with, you
+can use that, but otherwise this is an opinionated guide for setting up your
+dev environment. Follow these steps and you'll end out with a functioning dev
+environment and be able to submit your first PR.
+
+This guide assumes at least a passing familiarity with
+[Git](https://git-scm.com/), and that you already have it installed and
+configured.
+
+## `pyenv`, Virtual Environments, and You
+
+We recommend [pyenv](https://github.com/pyenv/pyenv), since it allows
+installing multiple different Python versions, which is important for testing
+Salt across all the versions of Python that we support.
+
+### On Linux
+
+Install pyenv: 
+
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    export PATH="$HOME/.pyenv/bin:$PATH"
+    git clone https://github.com/pyenv/pyenv-virtualenv.git $(pyenv root)/plugins/pyenv-virtualenv
+
+### On Mac
+
+Install pyenv using brew:
+
+    brew update
+    brew install pyenv
+    brew install pyenv-virtualenv
+
```

Comment:
**Suggestion (non-blocking):** We might want to ask @twangboy to write the section about how to set up a Windows environment.

Reply:
Maybe open a separate issue for this chore and you can close out this issue.